src/main/include/utils/State.h

```diff
@@ -0,0 +1,55 @@
+#ifndef STATE_H
+#define STATE_H
+
+#include "Constants.h"
+
+enum class RobotState {
+    Intaking,
+    ScoringSpeaker,
+    ScoringAmp,
+    Stowing,
+    Idling
+};
+
+class StateManager {
```

Comment:
Are there concerns about preventing certain actions based on the robot definition? Might need to wrap some of these in `#ifdef`